src/shop.service.ts

```diff
@@ -5,128 +5,132 @@ import { Observable, BehaviorSubject, ReplaySubject } from 'rxjs/Rx';
+    vegetarian: boolean;
+    local: boolean;
   };
 
 
   //
   // define where this shop is available for collect
-  collect?:{
+  collect?: {
     streetAdress: string;
     postalCode: string;
-    geo:{
-      lat:number;
-      lng:number;
+    geo: {
+      lat: number;
+      lng: number;
     }
   }
 
   //
   // where shop is located
-  address:{
+  address: {
     // an other place where things are stored
-    depository:string;
+    depository: string;
     name: string;
     floor: string;
     phone: string;
     streetAdress: string;
     region: string;
     postalCode: string;
-    geo:{
-      lat:number;
-      lng:number;
+    geo: {
+      lat: number;
+      lng: number;
     }
   };
 
   //
   // this shop belongsTo a category
-  catalog:any;
+  catalog: any;
 
   //
   // answer question about your shop
-  faq?:[{
-    q:string;
-    a:string;
-    updated:Date;
+  faq?: [{
+    q: string;
+    a: string;
+    updated: Date;
   }];
 
-  available:{
-    active:boolean;
-    from:Date,
-    to:Date,
-    weekdays:[number],
-    comment:string
+  available: {
+    active: boolean;
+    from: Date,
+    to: Date,
+    weekdays: [number],
+    comment: string
   };
 
-  discount:{
-    amount:number;
-    threshold:number;
-    active:boolean;
+  discount: {
+    amount: number;
+    threshold: number;
+    active: boolean;
   };
 
-  info:{
+  info: {
     //
     // requiere a detailled email for order preparation
-    detailledOrder:boolean;
-    active:boolean;
-    comment:{type: String}
+    detailledOrder: boolean;
+    active: boolean;
+    comment: { type: String }
   };
 
   //
   // type Date on pending, set true on active, false on deleted
-  status:any;
+  status: any;
   // secret value for the business model
   // - > is available/displayed for shop owner and admin ONLY
   // - > is saved on each order to compute bill
-  account:{
-    fees?:number;
-    tva:{
-      number?:number,
-      fees?:number
+  account: {
+    fees?: number;
+    tva: {
+      number?: number,
+      fees?: number
     },
-    updated:Date;
+    updated: Date;
   };
-  owner:any;
-  scoring:{
-    weight:number;
-    orders:number;
-    issues:number;
-    score:number;
+  owner: any;
+  scoring: {
+    weight: number;
+    orders: number;
+    issues: number;
+    score: number;
   };
-  created:Date;
+  created: Date;
 
   //
   // Object methods
-  constructor() {
-    let defaultShop = {
-      url:'',
-      photo:{fg:''},
-      options:{},
-      available:{},
-      collect:{},
-      address:{},
-      info:{},
-      account:{},
-      faq:[]
+  constructor(json: any) {
+    if (json !== null) {
+      Object.assign(this, json);
+    } else {
+      let defaultShop = {
+        url: '',
+        photo: { fg: '' },
+        options: {},
+        available: {},
+        collect: {},
+        address: {},
+        info: {},
+        account: {},
+        faq: []
+      }
+      Object.assign(this, defaultShop);
```

Comment:
* tu peux enlever le test `if` et toujours faire `Object.assign(this, defaultShop, json||{});`
* c'est valable pour tous les classes et c'est IMPORTANT de vérifier :+1: